mirdata/core.py

```diff
@@ -386,17 +452,69 @@ class MultiTrack(Track):
 
     A multitrack class is a collection of track objects and their associated audio
     that can be mixed together.
-    A multitrack is iteslf a Track, and can have its own associated audio (such as
+    A multitrack is itself a Track, and can have its own associated audio (such as
     a mastered mix), its own metadata and its own annotations.
 
     """
 
-    def _check_mixable(self):
-        if not hasattr(self, "tracks") or not hasattr(self, "track_audio_property"):
-            raise NotImplementedError(
-                "This MultiTrack has no tracks/track_audio_property. Cannot perform mixing"
+    def __init__(
+        self,
+        mtrack_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata=None,
+    ):
+        """Multitrack init method. Sets boilerplate attributes, including:
+
+        - ``mtrack_id``
+        - ``_dataset_name``
+        - ``_data_home``
+        - ``_multitrack_paths``
+        - ``_multitrack_metadata``
+
+        Args:
+            mtrack_id (str): multitrack id
+            data_home (str): path where mirdata will look for the dataset
+            dataset_name (str): the identifier of the dataset
+            index (dict): the dataset's file index
+            metadata (dict or None): a dictionary of metadata or None
+
+        """
+        if mtrack_id not in index["multitracks"]:
+            raise ValueError(
+                "{} is not a valid mtrack_id in {}".format(mtrack_id, dataset_name)
             )
 
+        self.mtrack_id = mtrack_id
+        self._dataset_name = dataset_name
+
+        self._data_home = data_home
+        self._multitrack_paths = index["multitracks"][self.mtrack_id]
+        self._metadata = metadata
+
+        self._index = index
+        self.track_ids = self._index["multitracks"][self.mtrack_id]["tracks"]
+        #self.mtrack_ids = self._index["multitracks"].keys()
+
+
+    @property
+    def tracks(self):
+        return {t: Track(t, self._data_home, self._dataset_name, self._index, self._metadata) for t in self.track_ids}
```

Comment:
you're right - there's a bug here. I think we need to pass `_track_class` to the init of the `MultiTrack` class like we do in the `Dataset` class, and then here do 
`{t: self._track_class(t, self._data_home, self._dataset_name, self._index, self._metadata) for t in self.track_ids}`